src/ProgrammingModel.ts

```diff
@@ -19,11 +19,13 @@ export class ProgrammingModel implements coreTypes.ProgrammingModel {
     async getCapabilities(capabilities: WorkerCapabilities): Promise<WorkerCapabilities> {
         lockSetup();
 
-        if (isHttpStreamEnabled()) {
+        if (enableHttpStream) {
             const httpUri = await setupHttpProxy();
             capabilities.HttpUri = httpUri;
         }
 
+        Object.assign(capabilities, libraryCapabilities);
```

Comment:
Should also change the name of this parameter from just `capabilities` to `workerCapabilities`? Just so we are clearly distinguishing the two different capabilities.